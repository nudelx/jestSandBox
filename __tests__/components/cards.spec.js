import React from 'react'
import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import EmptyCard from '../../src/components/emptyCard'
import MovieCard from '../../src/components/MovieCard'
import LoadingCard from '../../src/components/LoadingCard'
import Logo from '../../src/components/Logo'
import toJson from 'enzyme-to-json'
import enzymeSerializer from 'enzyme-to-json/serializer'
Enzyme.configure({ adapter: new Adapter() })

const movie = {
  data: {
    title: 'test title',
    opening_crawl: 'this is a crawl',
    director: 'BB-8',
    release_date: '1/1/1970'
  }
}

describe('all cards test', () => {
  it('empty card should match the snapshot', () => {
    const component = shallow(<EmptyCard />)
    expect(toJson(component)).toMatchSnapshot()
  })

  it('movie card should match the snapshot', () => {
    expect.addSnapshotSerializer(enzymeSerializer)
    const component = shallow(<MovieCard movie={movie} />)
    expect(component.props()).toMatchSnapshot()
    expect(component).toMatchSnapshot()
  })

  it('movie card should match the snapshot of the children with props', () => {
    expect.addSnapshotSerializer(enzymeSerializer)
    const component = shallow(<MovieCard movie={movie} />)
    expect(component.props()).toMatchSnapshot()
  })

  it('LoadingCard card should match the snapshot', () => {
    const component = shallow(<LoadingCard />)
    expect(toJson(component)).toMatchSnapshot()
  })
  it('Logo card should match the snapshot', () => {
    const component = shallow(<Logo />)
    expect(toJson(component)).toMatchSnapshot()
  })
})
