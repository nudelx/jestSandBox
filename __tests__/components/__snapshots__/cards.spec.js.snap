// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all cards test LoadingCard card should match the snapshot 1`] = `
<div>
  <div
    className="movie-title"
  >
    Loading...
  </div>
</div>
`;

exports[`all cards test Logo card should match the snapshot 1`] = `
<div
  className="sw-logo"
/>
`;

exports[`all cards test empty card should match the snapshot 1`] = `
<div>
  <div
    className="movie-title"
  />
  <div
    className="movie-data-empty"
  />
</div>
`;

exports[`all cards test movie card should match the snapshot 1`] = `
Object {
  "children": Array [
    <Logo />,
    <div
      className="movie-title"
    >
      test title
    </div>,
    <div
      className="movie-data"
    >
      <div
        className="animation"
      >
        this is a crawl
      </div>
    </div>,
    <div
      className="movie-data-bottom"
    >
      <div>
        Director: 
        BB-8
         
      </div>
      <div>
        1/1/1970
         
      </div>
    </div>,
  ],
}
`;

exports[`all cards test movie card should match the snapshot 2`] = `
<div>
  <Logo />
  <div
    className="movie-title"
  >
    test title
  </div>
  <div
    className="movie-data"
  >
    <div
      className="animation"
    >
      this is a crawl
    </div>
  </div>
  <div
    className="movie-data-bottom"
  >
    <div>
      Director: 
      BB-8
       
    </div>
    <div>
      1/1/1970
       
    </div>
  </div>
</div>
`;

exports[`all cards test movie card should match the snapshot of the children with props 1`] = `
Object {
  "children": Array [
    <Logo />,
    <div
      className="movie-title"
    >
      test title
    </div>,
    <div
      className="movie-data"
    >
      <div
        className="animation"
      >
        this is a crawl
      </div>
    </div>,
    <div
      className="movie-data-bottom"
    >
      <div>
        Director: 
        BB-8
         
      </div>
      <div>
        1/1/1970
         
      </div>
    </div>,
  ],
}
`;
